### ISSUE TITLE
v9.0.0 issue with ?fields=*.*.* API parameter

### ISSUE THREAD

**Issue by NecoHorne** [directus/directus#9590]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

My model has a multiple files attribute. To get the file details / nested details we used the `?fields=*.*.*` which returned all the file objects details. 

ex: 

```
 "images": [
        {
          "directus_files_id": {
            "id": "a6319ac9-c363-45c4-8980-73e1e1a7917f",
            "title": "798 8709 I1.jpg",
            "description": null,
            "filename_disk": "a6319ac9-c363-45c4-8980-73e1e1a7917f.jpg",
            "filename_download": "798_8709_I1.jpg",
            "type": "image/jpeg",
            "filesize": 90895,
            "width": 1024,
            "height": 768,
            "charset": null,
            "duration": null
          }
        },
        ....
```

After upgrading to V9.0.0 the api returned null for all the nested `directus_files_id` objects.

reverted back to rc101 which fixed the issue but caused other issues with roles after migrating the db down. But that is another bug. 

### To Reproduce

Create a data model with multiple images in rc and upgrade to v9.0.0 and use the `?fields=*.*.*` parameter on the collection API. 

### Errors Shown

none

### What version of Directus are you using?

9.0.0

### What version of Node.js are you using?

14.16.0

### What database are you using?

MySQL

### What browser are you using?

Chrome

### What operating system are you using?

Linux 4.14.248-189.473.amzn2.x86_64

### How are you deploying Directus?

Amazon Elastic Beanstalk

**Comment by nazariydj** [comment]
i had a similar problem with m2m fields, but it was due to the deep filter. i think there are some breaking changes (it’s working before).

**Comment by joselcvarela** [comment]
Hello @NecoHorne 
Do this happens too if you try a fresh installation?

___

> In order to have better help, it is advised to share a [schema snapshot](https://docs.directus.io/reference/cli/#snapshot-the-data-model)

**Comment by NecoHorne** [comment]
Hi @joselcvarela I have not tried a fresh installation. The project that experienced the issue is currently serving data to a production site so to get the images I reverted back to RC 101.

link to gist of the [snapshot](https://gist.github.com/NecoHorne/55f57824e68083c9baaabfce1c2eb856)

**Comment by josias-r** [comment]
Experiencing the same issue. Seems to be an issue related to permissions, as when I use an Admin token for the same query, I can get the correct data.

**Comment by rijkvanzanten** [comment]
I'm fairly sure this was caused by the same underlying issue as #9578 which was fixed in https://github.com/directus/directus/pull/9726. I'll close this for now, but we can reopen if the issue wasn't fixed 👍🏻

**Comment by NecoHorne** [comment]
> I'm fairly sure this was caused by the same underlying issue as #9578 which was fixed in #9726. I'll close this for now, but we can reopen if the issue wasn't fixed 👍🏻

Fixed in 9.0.1 thanks @rijkvanzanten 💪
